### PR TITLE
Add tag mapping and validation

### PR DESCRIPTION
**Pull request by ethanwharris**
# Before submitting

- [ ] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [x] Did you make sure to update the docs?
- [ ] Did you write any new necessary tests?

## What does this PR do?

Fixes # (issue).

## PR review

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

**Review by SeanNaren — APPROVED** [review]

**Review by justusschock — APPROVED** [review]

**Comment by Borda** [comment]
Mind explaining why the dictionary? 🐰

File: .actions/helpers.py [hunk 1]
```diff
@@ -344,7 +348,21 @@ def _get_card_item_cell(path_ipynb: str, path_meta: str, path_thumb: Optional[st
         if dirname != ".notebooks":
             meta["tags"].append(dirname)
 
-        meta["tags"] = ",".join(meta["tags"])
+        tags = []
```
Comment (Borda) [inline .actions/helpers.py:351]: Valid tags or all tags?

**Comment by ethanwharris** [comment]
@Borda Just to map the folder names to proper tag names. So instead of `course_UvA-DL` the tage will be `UvA DL Course`, similarly for lightning examples.

File: .actions/helpers.py [hunk 2]
```diff
@@ -344,7 +348,21 @@ def _get_card_item_cell(path_ipynb: str, path_meta: str, path_thumb: Optional[st
...
+        invalid_tags = []
+        for tag in meta["tags"]:
+            if " " in tag:
+                invalid_tags.append(tag)
+                continue
+            tags.append(HelperCLI.TAGS.get(tag, tag))
+
+        if invalid_tags:
+            raise ValueError(
+                f"The following tags are invalid: {','.join(invalid_tags)}. Tags should not contain spaces, use a "
```
Comment (Borda) [inline .actions/helpers.py:361]: Why not, is it some docs issue?
If so why just not replace all spaces by -

Comment (ethanwharris) [reply .actions/helpers.py:361]: Yeah, it only supports dashes. We could just replace spaces, I'll change it to that

File: .actions/helpers.py [hunk 3]
```diff
@@ -148,6 +148,10 @@ class HelperCLI:
     REQUIREMENTS_FILE = "requirements.txt"
     PIP_ARGS_FILE = "pip_arguments.txt"
     META_PIP_KEY = "pip__"
+    TAGS = {
+        "course_UvA-DL": "UvA-DL-Course",
```
Comment (Borda) [inline .actions/helpers.py:152]: @ethanwharris where is the name with spaces instead of the -
Maybe I just don't see it =)

Comment (ethanwharris) [reply .actions/helpers.py:152]: Haha, the dashes get turned into spaces in the tag view thing. So `course_UvA-DL` becomes `course_UvA DL` and `UvA-DL-Course` becomes `UvA DL Course`.

File: .actions/helpers.py (hunk 2 above)
Comment (ethanwharris) [reply .actions/helpers.py]: Done :smiley:

File: .actions/helpers.py (hunk 3 above)
Comment (Borda) [reply .actions/helpers.py:152]: I see so lest add this as comment to the code

File: .actions/helpers.py [hunk 4]
```diff
@@ -344,6 +348,7 @@ def _get_card_item_cell(path_ipynb: str, path_meta: str, path_thumb: Optional[st
         if dirname != ".notebooks":
             meta["tags"].append(dirname)
 
+        meta["tags"] = [HelperCLI.TAGS.get(tag, tag).replace(" ", "-") for tag in meta["tags"]]
```
Comment (Borda) [inline .actions/helpers.py:351]: looks much nicer :D

**Review by Borda — APPROVED** [review]

File: course_UvA-DL/activation-functions/.meta.yml [hunk 5]
```diff
@@ -3,6 +3,7 @@ author: Phillip Lippe
 created: 2021-08-27
 updated: 2021-08-27
 license: CC BY-SA
+build: 0
```
Comment (Borda) [inline course_UvA-DL/activation-functions/.meta.yml:6]: I think none of these builds is needed as tags are not part of notebook rendering

**Pull request merged**